### PR TITLE
New package: sq-0.42.0

### PR DESCRIPTION
**Pull request by icp1994**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### Local build testing
- I built this PR locally for my native architecture: **x86_64**

### CODE REVIEW

File: srcpkgs/sq/template [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Template file for 'sq'
+pkgname=sq
+version=0.40.0
+revision=1
+archs="x86_64* aarch64*"
```
Comment (Piraty) [inline srcpkgs/sq/template:5]: what's the reasoning to limit archs here?

Comment (icp1994) [reply srcpkgs/sq/template:5]: It doesn't build in 32-bit systems, at least [in CI](https://github.com/void-linux/void-packages/actions/runs/5641324418)

Comment (Piraty) [reply srcpkgs/sq/template:5]: why preclude ppc64  and riscv64 then?

Comment (icp1994) [reply srcpkgs/sq/template:5]: Void doesn't officially support ppc anymore, does it? And I have no way to verify if it actually builds for riscv so I didn't include it.

**Comment by leahneukirchen** [comment]
This fixes compilation on 32-bit, feel free to upstream:

```
--- sq-0.42.0/libsq/ast/render/range.go.orig
+++ sq-0.42.0/libsq/ast/render/range.go
@@ -27,7 +27,7 @@
                if rr.Limit == -1 {
                        // MySQL requires a LIMIT if OFFSET is used. Therefore
                        // we make the LIMIT a very large number
-                       limit = fmt.Sprintf("LIMIT %d", math.MaxInt64)
+                       limit = fmt.Sprintf("LIMIT %d", uint64(math.MaxInt64))
                }
        }
```

**Comment by icp1994** [comment]
Thank you!

File: srcpkgs/sq/template (hunk 1 above)
Comment (Piraty) [reply srcpkgs/sq/template]: yet there's no reason to spoil the fun for those who build on their own.

**Comment by neilotoole** [comment]
Hi, I'm the [`sq`](https://github.com/neilotoole/sq) author. @icp1994's PR has been merged, and released as [`sq v0.42.1`](https://github.com/neilotoole/sq/releases/tag/v0.42.1). Let me know if there's anything else I can do to help.

**Pull request merged**